build/stf-run-ci/tasks/create_catalog.yml

```diff
@@ -151,7 +185,25 @@
       dest: "{{ base_dir }}/working/service-telemetry-framework-index/index.yaml"
 
   - name: Build service-telemetry-framework-index
-    ansible.builtin.command: oc start-build -n "{{ namespace }}" service-telemetry-framework-index --wait --from-dir {{ base_dir }}/working/service-telemetry-framework-index
+    ansible.builtin.command: oc start-build -n "{{ namespace }}" service-telemetry-framework-index --from-dir {{ base_dir }}/working/service-telemetry-framework-index
+    ignore_errors: true
+
+  - name: Kill the first build
```

Comment:
When a BuildConfig is created, that automatically starts a build?

Reply:
Looking at this again, I am also confused why 2 start-builds are required here. As Chris noted, the deletion of the build is because the `oc create-build` kicks off a build without content, so it fails. I'd not expect the same thing to happen on the first build of an `oc start-build`.

Reply:
I'll have  to look into it again

Reply:
I just did some manual testing here with the contents and it looks like the builds with `oc start-build` work fine. There is also no automatic trigger of builds on the index when I fresh create the BuildConfig from a template.

I also noticed something that is no longer required, so I'll provide a suggested change there.

From my history:

```
 1024  oc delete buildconfig service-telemetry-framework-index
 1025  oc get builds
 1026  oc create -f - <<EOF
apiVersion: build.openshift.io/v1
kind: BuildConfig
metadata:
  annotations:
    openshift.io/generated-by: stf-run-ci
  labels:
    build: service-telemetry-framework-index
  name: service-telemetry-framework-index
  namespace: "service-telemetry"
spec:
  failedBuildsHistoryLimit: 5
  nodeSelector: null
  output:
    to:
      kind: ImageStreamTag
      name: service-telemetry-framework-index:latest
  postCommit: {}
  resources: {}
  runPolicy: Serial
  source:
    dockerfile: |
      # The base image is expected to contain
      # /bin/opm (with a serve subcommand) and /bin/grpc_health_probe

      FROM quay.io/openshift/origin-operator-registry:4.13

      COPY --chmod=666 index.yaml /configs/

      RUN mkdir /tmp/auth/
      # we need the contents of the mounted build volume from secret placed into config.json
      RUN cp /opt/app-root/auth/.dockerconfigjson /tmp/auth/config.json
      RUN DOCKER_CONFIG=/tmp/auth /bin/opm --skip-tls-verify render image-registry.openshift-image-registry.svc:5000/service-telemetry/service-telemetry-operator-bundle@sha256:6cb291fc597b8f2e956a04f3d58aa6978d025c3ad63eac7309cb3453cd4a3d38 image-registry.openshift-image-registry.svc:5000/service-telemetry/smart-gateway-operator-bundle@sha256:3a390a829460b3c9523ffcf0ab881c3d56ba450b770abe48873596b7bdead7f5 --output=yaml >> /configs/index.yaml

      ENTRYPOINT ["/bin/opm"]
      CMD ["serve", "/configs"]
      # Set DC-specific label for the location of the DC root directory
      # in the image
      LABEL operators.operatorframework.io.index.configs.v1=/configs
    type: Dockerfile
  strategy:
    dockerStrategy:
      from:
        kind: ImageStreamTag
        name: "ose-operator-registry:4.13"
      volumes:
      - mounts:
        - destinationPath: /opt/app-root/auth
        name: pull-secret
        source:
          secret:
            defaultMode: 420
            secretName: service-telemetry-framework-index-dockercfg
          type: Secret
    type: Docker
  successfulBuildsHistoryLimit: 5
EOF

 1031  cd ~/src/github.com/infrawatch/service-telemetry-operator
 1032  ls
 1033  cd build/
 1034  ls working/
 1035  cat working/service-telemetry-framework-index/index.yaml 
 1036  oc start-build -n "service-telemetry" service-telemetry-framework-index --from-dir ./working/service-telemetry-framework-index
 1039  oc get build service-telemetry-framework-index-1 -w


NAME                                  TYPE     FROM             STATUS     STARTED         DURATION
service-telemetry-framework-index-1   Docker   Binary@4e1bd7e   Complete   3 minutes ago   26s
```